### ISSUE TITLE
Users shouldn't be forced to run "mixer" commands at the root level of their mix workspace

### ISSUE THREAD

**Issue by bktan8** [clearlinux/mixer-tools#519]
Right now, if I run the "mixer" command in any subdirectory of my mixer workspace, I get this error: 

```
2018/12/11 23:58:43 Warning: Using FORMAT value from /usr/share/defaults/swupd/format
2018/12/11 23:58:43 ERROR: open /home/clear/mix2/local-rpms/builder.conf: no such file or directory
```
and a "mixer.state" file gets generated at the location I was at.

IMO, it's not user friendly to force users to have to go their mixer root workspace before executing any "mixer" command.  Users will want to navigate freely within their mix workspace and execute any "mixer" command.

**Comment by rchiossi** [comment]
Are you sure you are running the latest version of mixer? This changed in mixer 5.5

**Comment by bktan8** [comment]
I'm running 5.5.1.  See example below:

```
clear@clr-os ~ $ mixer init 
Initializing mix version 10 from upstream version 26760
Adding bundle "os-core" from upstream bundles
Adding bundle "os-core-update" from upstream bundles
Adding bundle "bootloader" from upstream bundles
Adding bundle "kernel-native" from upstream bundles

clear@clr-os ~ $ mixer --version
Mixer 5.5.1

clear@clr-os ~ $ cd local-bundles/

clear@clr-os ~/local-bundles $ mixer bundle list
2018/12/12 17:54:21 Warning: Using FORMAT value from /usr/share/defaults/swupd/format
2018/12/12 17:54:21 ERROR: open /home/clear/local-bundles/builder.conf: no such file or directory

clear@clr-os ~/local-bundles $ ls
mixer.state
```

**Comment by rchiossi** [comment]
Ah ok. I thought you meant `mixer` as in the root command without any parameters, which got fixed in 5.5.

First, `mixer.state` being created there is a known bug and will be fixed. So disregard that for now.

So my understanding is that you want something like `git`, which is able to look for the its metafiles on top directories. The problem is that unlike git, mixer does not have self-contained workspace. In your example, local-bundles is part of your workspace, but It could be anywhere in your system since it is determined by the `LOCAL_BUNDLE_DIR` variable in builder.conf. The same is true for most of the files mixer uses, including `mixbundles` which is the one mixer uses to determine the output of `mixer bundle list`. A good example of spread out workspace is the devops environment. 

Right now, mixer needs to be able to find `builder.conf` to run. I know that is not what you want since it is not convenient, but you can provided `--config` to run it from any directory.

**Comment by reaganlo** [comment]
This has been fixed.
